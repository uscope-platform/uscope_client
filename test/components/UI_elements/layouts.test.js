// Copyright 2021 Filippo Savi
// Author: Filippo Savi <filssavi@gmail.com>
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import {ThemeProvider} from "styled-components";

import {
    ApplicationLayout,
    BlockLayout,
    ColorTheme,
    SidebarBlockLayout,
    SidebarContentLayout, SidebarLayout
} from "../../../src/components/UI_elements";


test('block_layout', () => {
    const tree = renderer.create(
        <ThemeProvider theme={ColorTheme}>
            <BlockLayout />
        </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})

test('block_layout_centered', () => {
    const tree = renderer.create(
        <ThemeProvider theme={ColorTheme}>
            <BlockLayout centered/>
        </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})

test('sidebar_block_layout', () => {
    const tree = renderer.create(
        <ThemeProvider theme={ColorTheme}>
            <SidebarBlockLayout />
        </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})

test('sidebar_block_layout padding ', () => {
    const tree = renderer.create(
        <ThemeProvider theme={ColorTheme}>
            <SidebarBlockLayout padding={'1rem'}/>
        </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})


test('sidebar_content_layout', () => {
    const tree = renderer.create(
        <ThemeProvider theme={ColorTheme}>
            <SidebarContentLayout />
        </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})



test('sidebar_content_layout application', () => {
    const tree = renderer.create(
        <ThemeProvider theme={ColorTheme}>
            <SidebarContentLayout application/>
        </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})



test('sidebar_content_layout peripheral', () => {
    const tree = renderer.create(
        <ThemeProvider theme={ColorTheme}>
            <SidebarContentLayout peripheral/>
        </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})

test('application_layout', () => {
    const tree = renderer.create(
        <ThemeProvider theme={ColorTheme}>
            <ApplicationLayout/>
        </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})


test('application_layout sidebar', () => {
    const tree = renderer.create(
        <ThemeProvider theme={ColorTheme}>
            <ApplicationLayout sidebarNeeded={true}/>
        </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})

test('sidebar layout', () => {
    const tree = renderer.create(
        <ThemeProvider theme={ColorTheme}>
            <SidebarLayout/>
        </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})








