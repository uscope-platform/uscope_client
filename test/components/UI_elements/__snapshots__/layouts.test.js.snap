// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application_layout 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 2fr 8fr 2fr;
  grid-gap: 1.5rem;
  margin: 1rem 0 0 0;
  grid-auto-rows: auto;
  height: 96.5vh;
  grid-template-columns: 2fr 10fr;
}

<div
  className="c0"
/>
`;

exports[`application_layout sidebar 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 2fr 8fr 2fr;
  grid-gap: 1.5rem;
  margin: 1rem 0 0 0;
  grid-auto-rows: auto;
  height: 96.5vh;
}

<div
  className="c0"
/>
`;

exports[`block_layout 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 1fr;
  grid-auto-rows: max-content;
  grid-row-gap: 1em;
  background-color: #122964;
  border-radius: 1rem;
  padding: 1rem;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  width: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

<div
  className="c0"
/>
`;

exports[`block_layout_centered 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 1fr;
  grid-auto-rows: max-content;
  grid-row-gap: 1em;
  background-color: #122964;
  border-radius: 1rem;
  padding: 1rem;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  width: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-left: auto;
  margin-right: auto;
}

<div
  className="c0"
/>
`;

exports[`sidebar layout 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 1fr;
  grid-auto-rows: auto;
  background-color: #122964;
  margin-left: 1rem;
}

<div
  className="c0"
/>
`;

exports[`sidebar_block_layout 1`] = `
.c0 {
  border-radius: 1rem;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 0.2rem 0 0.4rem 0.6rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #1d7097;
}

<div
  className="c0"
/>
`;

exports[`sidebar_block_layout padding  1`] = `
.c0 {
  border-radius: 1rem;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 1rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #1d7097;
}

<div
  className="c0"
/>
`;

exports[`sidebar_content_layout 1`] = `
.c0 {
  display: grid;
  height: 100vh;
  grid-template-columns: 1fr;
  grid-gap: 0.5rem;
  margin-left: 0.8rem;
  margin-right: 0.8rem;
  justify-items: center;
}

<div
  className="c0"
/>
`;

exports[`sidebar_content_layout application 1`] = `
.c0 {
  display: grid;
  height: 100vh;
  grid-template-columns: 1fr;
  grid-auto-rows: auto 10fr auto;
  grid-gap: 0.5rem;
  margin-left: 0.8rem;
  margin-right: 0.8rem;
  justify-items: center;
}

<div
  className="c0"
/>
`;

exports[`sidebar_content_layout peripheral 1`] = `
.c0 {
  display: grid;
  height: 100vh;
  grid-template-columns: 1fr;
  grid-auto-rows: auto auto 10fr auto;
  grid-gap: 0.5rem;
  margin-left: 0.8rem;
  margin-right: 0.8rem;
  justify-items: center;
}

<div
  className="c0"
/>
`;
